File: 00_ETC_PY/29823.py
n = int(input())
str = input()

res = 0
cntE = cntS = cntW = cntN = 0

for i in range(n):
    if str[i] == 'S':
        cntS += 1
    elif str[i] == 'N':
        cntN += 1
    elif str[i] == 'E':
        cntE += 1
    elif str[i] == 'W':
        cntW += 1

res = abs(cntE - cntW) + abs(cntS - cntN)
print(res)

File: 00_ETC/29823.cpp
#include <iostream>
using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);

    int n;
    string str;
    cin >> n;
    cin >> str;

    int res = 0, cntE = 0, cntS = 0, cntW = 0, cntN = 0;
    for (int i = 0; i < n; ++i) {
        if (str[i] == 'E') {
            ++cntE;
        } else if (str[i] == 'S') {
            ++cntS;
        } else if (str[i] == 'W') {
            ++cntW;
        } else if (str[i] == 'N') {
            ++cntN;
        }
    }
    res = abs(cntE - cntW) + abs(cntS - cntN);
    cout << res << endl;
    return 0;
}
